fakemails/generate-fake-mail.py

```diff
@@ -0,0 +1,216 @@
+def copy_fake_tempates(account, fake_dir):
+    try:
+        if os.path.exists(f"/var/mail/{account}/Maildir/cur") == True:
+            uid, gid = get_user_infos(account)
+            subprocess.run(["/bin/bash", "./rsync.sh", account, uid, gid, fake_dir])
+        else:
+            print(f"Default folder for {account} does not exists")
+    except Exception as e:
+        print(f"Failed to retrieve account information for {account}: {e}")
+
+def exec_copy(accounts, fake_dir):
+    num_processes = 5
+
+    with ThreadPoolExecutor(max_workers=num_processes) as executor:
+        futures = [executor.submit(copy_fake_tempates, account, fake_dir) for account in accounts]
+        
+        for future in futures:
+            future.result()
+
+if __name__ == '__main__':
+    tdates = [1609459200, 1640995200, 1672531200, 1704067200, round(time.time())]
+    templates = {}
+    files_size = 0
+    accounts = []
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--accountsfile', type=str, help='List of users accounts', default=f'{os.getcwd()}/accounts')
+    parser.add_argument('--templatedir', type=str, help='Template directory', default=f'{os.getcwd()}/mail-templates')
+    parser.add_argument('--fakedir', type=str, help='Destination directory for fake emails', default=f'{os.getcwd()}/fake-mails')
+    args = parser.parse_args()
+
+    if os.path.exists(args.fakedir) == False:
+        os.mkdir(args.fakedir)
+
+    if os.path.isfile(args.accountsfile) == True:
+        with open(args.accountsfile, 'r') as f:
+            accounts = [line.strip() for line in f.readlines() if line.strip()]
+    else:
+        print(f"File {args.accountsfile} does not exists")
+        os._exit(1)
+
+    for dirpath, _, filenames in os.walk(args.templatedir):
+        for file in filenames:
+            with open(f"{args.templatedir}/{file}", 'r') as f:
+                templates[file] = f.read()
+                files_size += len(templates[file])
+
+    total_size = input("What is the total size that you would like to generate for each account? You can use MB, GB or TB : ")
+    bsize = convert_total_size_to_bytes(total_size)
+
+    if bsize < 31457280:
+        print("Total size needs to be greater than 30MB.")
+        os._exit(1)
+
+    total_files = round(bsize / files_size)
+    total_iterations = round(total_files / 4)
+    print(f"The script will generate {total_iterations * 4} files for each template based on {total_size} of fake emails for each user on the list. Emails will be generated from the beginning of the year 2021 up to the present moment.")
+    
+    for key, value in templates.items():
+        for i, t in enumerate(tdates[0:4]):
+            for z in range(total_iterations):
+                content, dovecot_message_id, timestamp = generate_fake_mail_content("admin", t, tdates[i+1: i+2][0], value)
+                create_fake_mail(content, dovecot_message_id, timestamp, args.fakedir)
+
+    exec_copy(accounts, args.fakedir)
+
```

Comment:
Using `round(time.time())` for `tdates` is not ideal as it can lead to non-deterministic behavior in your tests or in the execution of your script. Consider using a fixed end date or handling this through configuration.
```suggestion
tdates = [1609459200, 1640995200, 1672531200, 1704067200, 1704067200]
```